pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -392,6 +393,13 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 	con := newXdsConnection(peerAddr, stream)
 	defer close(con.doneChannel)
 
+	// generate a connection lifetime that is +- 10% of MaxConnLifetime to avoid
```

Comment:
I have not looked at the detail here - so obviously may be missing some thing here. But gRPC already supports max-connection-age with jitter - should n't that be used instead of re-implementing it here?
Here are java docs https://github.com/grpc/grpc-java/blob/master/netty/src/main/java/io/grpc/netty/NettyServerBuilder.java#L375 and similar thing in go here https://github.com/grpc/grpc-go/blob/04ea82009cdb9ecdefc6289f4c93ec919a10b3b6/internal/transport/http2_server.go#L186

Reply:
Neat! I actually didn't know that there was native support for this. It looks like this PR will be supplanted by a different one that uses max-connection-age, but at least I learned something new.